### PR TITLE
LinkedList is done

### PR DESCRIPTION
**Pull request by yermilov-m**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -1,47 +1,144 @@
...
+            add(value);
+        } else {
+            if (index == 0) {
+                newNode = new Node<>(null, value, head);
+                head.prev = newNode;
+                head = newNode;
+            } else {
+                checkIndex(index);
+                Node<T> prevNode = getNode(index - 1);
+                Node<T> nextNode = getNode(index);
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/MyLinkedList.java:41]: call **getNode** once
```suggestion
                Node<T> nextNode = getNode(index);
```

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -1,47 +1,144 @@
...
+        if (size == 0) {
+            newNode = new Node<>(null, value, null);
+            head = newNode;
+        } else if (size == 1) {
+            newNode = new Node<>(head, value, null);
+            head.next = newNode;
+        } else {
+            newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+        }
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/MyLinkedList.java:23]: try to simplify this method. Create new Node once before IF. Have one IF condition and one ELSE

**Review by Rommelua — CHANGES_REQUESTED** [review]
https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#create-a-separate-method-unlinknode-node

image

Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -1,47 +1,135 @@
...
+        size--;
+        return currentNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(currentNode.value, object)) {
+                remove(i);
```
Comment (aaananas) [inline src/main/java/core/basesyntax/MyLinkedList.java:77]:
```suggestion
                unlink(currentNode);
```

**Review by aaananas — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Cat.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package core.basesyntax;
+
+import java.util.Objects;
+
+public class Cat {
```
Comment (andriy-shulepa) [inline src/main/java/core/basesyntax/Cat.java:5]: this is an unnecessary change. remove it

**Review by andriy-shulepa — CHANGES_REQUESTED** [review]

**Review by andriy-shulepa — APPROVED** [review]
